### ISSUE TITLE
CWM theme editor refactor

### ISSUE THREAD

**Issue by RubenSmn** [Greenstand/treetracker-web-map-client#1604]
### Problem

Currently we're using the [theme from the themeContext](https://github.com/Greenstand/treetracker-web-map-client/blob/main/src/context/themeContext.js) to render our inputs for the editor this does not scale too well.

We currently have 3 types of inputs, 'text', 'select', 'color'. How we determine which input to render is not clear :cry:. The way the data flows from the `theme` constant to the `input` being chosen is hard to follow.

### Approach

We're going to take a more config based approach. The following code snippet is from [this project](https://github.com/RubenSmn/spotify-banner-creator/blob/main/src/constants/config.ts) which I build almost a year ago.

```javascript
const config = {
  fontSize: {
    displayText: 'Size',
    input: PROPTYPES.SLIDER, // 'slider'
    min: 3,
    max: 5,
    step: 0.2,
    defaultValue: 4,
    unit: 'rem',
  },
};
```

As you can see the **idea** is that we can add a property to the `config`, define the requirements for that specific property, such as what kind of input it needs to be e.g. 'slider', 'text', the additional values according to that input type.

---

This new flow will improve developer experience by being able to configure the sections e.g. colors, typography in a single config file and improve the UI/UX for the end user by introducing more straight forward inputs based on the value type.

**Comment by RubenSmn** [comment]
Please let me know any taughts you have on this!

**Comment by dadiorchen** [comment]
@RubenSmn if we use custom syntax for the style config, how do we use it when we render the web map?

**Comment by dadiorchen** [comment]
@RubenSmn this is about how to define the editor so people can set up things on it, right? If for a person how directly edit the theme code, this is unnecessary, right?

**Comment by RubenSmn** [comment]
@dadiorchen this is about how the sections and inputs are rendered in the editor itself. So with the new setup we are able to setup new things better and we have more control over what type of input we use for a property. For example we have the `fontSize` and currently we have just a simple text input where you type the amount of pixels. With the new approach we can configure the `fontSize` to be adjusted by a slider with a standard unit of `px` and the **end user** does not need to know anything about what the unit needs to be, they can simply slide the slider to the size they want.

The structure of the output will still be the same, we will still use a **path** e.g. `palette.background.primary.main` to update the style in the actual theme object.

**Comment by dadiorchen** [comment]
So this will be two phases: 1. The template of the theme with the self-defined input you proposed here. 2. Translate the theme to the final form that can be used in our client?

**Comment by RubenSmn** [comment]
Part 1 yes, part 2 is already in place and does not need changes.

**Comment by RubenSmn** [comment]
Resolved by #1619
